File: SRP/violations/Order.ts
export class Order {
}

export interface Logger {
    log(message: string): void;
}

export class OrderService {
    constructor(
        private db: unknown,   // persistence concern
        private logger: Logger // logging concern
    ) { }

    placeOrder(order: Order): void {
        // everything happens here
    }
}

File: SRP/violations/Order.cs
namespace SRP.Violations
{
    public class Order
    {
    }

    public class OrderService
    {
        private object db;          // persistence concern
        private ILogger logger;     // logging concern

        public void PlaceOrder(Order order)
        {
            // everything happens here
        }
    }

    // Sample logger abstraction for demonstration
    public interface ILogger
    {
        void Log(string message);
    }
}
